tensorflow/core/util/mkl_util.h

```diff
@@ -1137,13 +1490,30 @@ class MklDnnData {
     auto pd = memory::primitive_desc(md, *cpu_engine_);
     SetUsrMem(pd, data_buffer);
   }
+#endif
 
   /// A version of SetUsrMem with memory descriptor and tensor
   inline void SetUsrMem(const memory::desc& md, const Tensor* tensor) {
     CHECK_NOTNULL(tensor);
     SetUsrMem(md, GetTensorBuffer(tensor));
   }
 
+#ifdef ENABLE_MKLDNN_V1
+  /// A version of function to set user memory type that accepts memory
+  /// descriptor directly, instead of accepting dimensions and format. This
+  /// function is more generic than the one above, but the function above is
+  /// sufficient in most cases.
+  inline void SetUsrMem(const memory::desc& md, void* data_buffer = nullptr) {
```

Comment:
Since this has the same signature as the one in line 1489, please merge the functions together and make them share the comment documentation. There should also be a clean way to also merge the one in line 1521 altogether.

Reply:
Done